prowler/providers/azure/azure_provider.py

```diff
@@ -619,8 +694,10 @@ async def get_azure_identity():
                 logger.critical(
                     "It was not possible to retrieve any subscriptions, please check your permission assignments"
                 )
-                # TODO: add custom exception once we have the AzureException
-                raise SystemExit
+                raise AzureNoSubscriptionsError(
+                    file=os.path.basename(__file__),
+                    message="No subscriptions were found",
```

Comment:
The same as the above comment,
`No subscriptions were found, please check your permission assignments`

Reply:
Done